Skuntank G PL

saavanverma opened the pull request.

saavanverma commented:
https://github.com/user-attachments/assets/9f344a3f-0069-4e40-86aa-499fb7401e70


https://github.com/user-attachments/assets/9f97f5a9-8cb0-4f80-acd8-0875d67cb8f8

saavanverma commented:
Attack copied from Magmar FO

saavanverma commented:
(First video is the 2nd turn, where stadium owner is opponent)

the-epsd commented on ptcg-server/src/sets/set-platinum/skuntank-g.ts:
```diff
@@ -0,0 +1,129 @@
+        throw new GameError(GameMessage.POWER_ALREADY_USED);
+      }
+
+      if (player.active.cards[0] === this && player.active.specialConditions.length > 0) {
```
Skunktank G does not need to be in the Active for this ability to work, you need to player.forEachPokemon to find Skunktank G and check special conditions that way.

the-epsd commented on ptcg-server/src/sets/set-platinum/skuntank-g.ts:
```diff
@@ -0,0 +1,129 @@
+        throw new GameError(GameMessage.CANNOT_USE_POWER);
+      }
+
+      ADD_POISON_TO_PLAYER_ACTIVE(store, state, player, this);
```
You need to add checks to see if the active is a Pokémon SP or not before applying the Poison.

saavanverma commented:
Fixed, added the check for SP

"if (player.active.cards[0] === this && player.active.specialConditions.length > 0) {" addresses the text on the card that says "This power can’t be used if Skuntank is affected by a Special Condition".

saavanverma commented:
there's no way for it to be special conditioned on the bench is there?
